contracts/governance/Governor.sol

```diff
@@ -389,39 +433,50 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor, IERC721Receive
     }
 
     /**
-     * @dev Hook before execution is triggered.
+     * @dev Internal queuing mechanism. Can be overridden (with a super call) to add logic before or after the
+     * queuing.
+     *
+     * Emits a {IGovernor-ProposalQueued} event if the eta is not 0.
      */
-    function _beforeExecute(
-        uint256 /* proposalId */,
+    function _queue(
+        uint256 proposalId,
         address[] memory targets,
-        uint256[] memory /* values */,
+        uint256[] memory values,
         bytes[] memory calldatas,
-        bytes32 /*descriptionHash*/
+        bytes32 descriptionHash
     ) internal virtual {
-        if (_executor() != address(this)) {
-            for (uint256 i = 0; i < targets.length; ++i) {
-                if (targets[i] == address(this)) {
-                    _governanceCall.pushBack(keccak256(calldatas[i]));
-                }
-            }
+        _validateStateBitmap(proposalId, _encodeStateBitmap(ProposalState.Succeeded));
+
+        uint256 eta = _queueCalls(proposalId, targets, values, calldatas, descriptionHash);
+
+        if (eta == 0) {
+            revert GovernorQueueNotImplemented();
```

Comment:
This is clever, but it seems a little dangerous to assume that a timelock would never create an eta of 0. Our own TimelockController allows that, and in that case the call to schedule is required, it is not possible to go directly to execute.

We may have to return `(bool, uint)`.